sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryUtils.java

```diff
@@ -390,6 +394,35 @@ public static Row toBeamRow(GenericRecord record, Schema schema, ConversionOptio
     return Row.withSchema(schema).addValues(valuesInOrder).build();
   }
 
+  /**
+   * Convert to a Beam {@link Row} from an Avro {@link GenericRecord}, using the avro schema to
+   * differentiate between Avro {@link org.apache.avro.LogicalType}.
+   */
+  public static Row toBeamRow(
+      GenericRecord record,
+      org.apache.avro.Schema avroSchema,
+      Schema beamSchema,
+      ConversionOptions options) {
```

Comment:
I don't actually see this new function used anywhere outside of tests, are you adding it as a public API for general purpose Avro -> Row conversion? If so I think something like this change may be better in [`AvroUtils`](https://github.com/apache/beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/utils/AvroUtils.java). In fact, I think much of the existing Avro conversion code in this class should be moved there, or should defer to it.

@kennknowles why do we have so much Avro to Row code here rather than deferring to the schema conversion code, is it because this predates org.apache.beam.sdk.schemas?

Reply:
Yes - the goal was to add support to the public API, as I found I couldn't rely on the existing implementation of convertAvroFormat. I didn't realize there was an equivalent method in AvroUtils! I'll test it out and see if I can use that one.